src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -99,6 +151,17 @@ var Marathon = React.createClass({
         this.stopPolling();
       }
     }
+
+    var route = this.state.route;
```

Comment:
https://github.com/mesosphere/marathon/blob/325-make-resources-linkable_cleaned/src/main/resources/assets/js/components/Marathon.jsx#L242

If I close the AppModal by click on 'x', this sets the route to '#apps' or '#deployments'. But if I change the route url from '#apps/app1' to '#apps/app2' by hand in the address bar, then this should not be triggered.